dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxrs.3.2/src/com/ibm/ws/jaxrs21/providers/json/JsonBProvider.java

```diff
@@ -42,13 +48,44 @@
     
     public JsonBProvider(JsonbProvider jsonbProvider) {
         if(jsonbProvider != null) {
-            this.jsonb = jsonbProvider.create().build();            
-        } else {            
-            if(Boolean.getBoolean("com.ibm.ws.jaxrs.testing") || System.getProperty("user.dir").contains("jaxrstck")) {
-                this.jsonb = null;
-            } else {
-                throw new IllegalArgumentException("jsonbProvider can't be null");
+            this.jsonb = jsonbProvider.create().build();
+        } else {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "<init> called with null provider - looking up via META-INF/services/"
+                    + JsonbProvider.class.getName());
             }
+            try {
+            this.jsonb = AccessController.doPrivileged(new PrivilegedExceptionAction<JsonbProvider>(){
+
+                @Override
+                public JsonbProvider run() throws Exception {
+                    // first try thread context classloader
+                    Iterator<JsonbProvider> providers = ServiceLoader.load(JsonbProvider.class).iterator();
+                    if (providers.hasNext()) {
+                        return providers.next();
+                    }
+                    // next try this classloader
+                    providers = ServiceLoader.load(JsonbProvider.class, JsonBProvider.class.getClassLoader()).iterator();
+                    if (providers.hasNext()) {
+                        return providers.next();
+                    }
+                    
+                    // not good - but maybe we're in a test environment where there will be no
+                    // need for a JSON-B provider...
+                    if (Boolean.getBoolean("com.ibm.ws.jaxrs.testing") || System.getProperty("user.dir").contains("jaxrstck")) {
+                        return null;
```

Comment:
Need to add  || System.getProperty("user.dir").contains("javaeetck")  as Hank runs the ee tck not the jaxrs tck